### ISSUE TITLE
FYI. Worked OK on Dell Latitude 3378 

### ISSUE THREAD

**Issue by rogerclarkmelbourne** [kt315ua/dell-charger-emu-ds2502#1]
I didn't bother pulling any of the config pins low, and it worked OK with a Dell Latitude 3379. (Core i5)

i.e This labtop seems to accept the Arduino sending it the data for a 130W charger, even though I think it uses a lower wattage charger, as the laptop has the suppy current listed as 2.3A and the voltage as 19.5V, which I presume means it can operate from a 45W charger.

Edit

I did some more testing and for my laptop (Latitude 3379), the only part of the OneWire EEPROM which is read by the bios is the wattage value e.g. 045

If a lower value is used, the BIOS displays a warning showing the lower value e.g. 030   shows "30W" in the BIOS warning screen.

I tried putting "DC" instead of AC and this was ignored.
I tried setting the voltage and current values to "000"  and these were also ignored.
I tried clearing the serial number to spaces, and this didn't make any difference.

Setting the entire string to spaces apart from the wattage and voltage and current also made no difference and the laptop worked fine.

I also measured the current consumption of the laptop, with the battery removed, and it didn't seem any different using different values of wattage above 45 ( "045").

Reconnecting the battery and trying different wattages also didn't seem to make any difference to the current being drawn, while charging (and not running). Changing the current value e.g. 023 to 500 also made no difference, it always took about 750mA at 20V 

Note.
To do this testing I had to add a CRC16-ARC function from one of the other Dell charger projects, to dynamically create the 2 CRC bytes.
The official Arduino CRC16 didn't seem to give the correct values. Probably because there are multiple types of CRC and this application requires the ARC version, which didn't seem to be supported by the official Arduino CRC library

**Comment by KayTeeFive** [comment]
Hi, interesting investigations!
https://i.dell.com/sites/csdocuments/Shared-Content_data-Sheets_Documents/en/Latitude-3379-spec-sheet.pdf
From datasheet your laptop require 45w, so, it's good sign that it's works fine event with fake 130w ID's.

I thing laptops strongly limited for their power consumption, because in another way it will cause unexpected heat dissipation some where inside.
The same thing, i think, related for battery charging also.
Additionally, laptop batteries are smart. They talk to motherboard power controller chip and it's selecting charging current requested by battery.

As I know, this hardcoded ID's are from real chargers, and i take this code from [this place](https://github.com/orgua/OneWireHub/tree/main/examples/DS2502_DELLCHG).
So, using real PSU with power less than you are set by emu, it's on your own risk.

About pull-up resistor, it's just recommended. Because, we don't know is there pull-up resistor on laptop side. And it can depends on MCU.

**Comment by rogerclarkmelbourne** [comment]
Probably some other Dell laptops read more of the data than mine does, so it would be safer to use the reverse engineered values for each specific wattage of PSU, but I was interested to see if my laptop would accept different wattage values etc, because in the long term I will probably buy a pre-programmed OneWire memory chip aka "Dell decoder" from AliExpress, and 45W chip is harder to find. The 90W chip seems to be the cheapest and is available from more suppliers.
And I now know that the 90W chip should work with my laptop.


BTW.

The battery in my laptop is almost unusable. The battery its self does not appear to be an intelligent module, the battery changing circuit seems to be part of the motherboard.

A friend gave me the laptop but did not have the charger, and the screen has some problems and also the keyboard has some key faults.

So, I do not plan on buying a new battery, and instead will probably use this as a server or media player and always use external power.

**Comment by KayTeeFive** [comment]
Yep, this Arduino sketch is universal to check different Dell PSU ID's with different wattage.
By the way, with dell batteries is one thing. Periodically they add different communication protocols to prevent using a fake or aftermarket non-original batteries. In this case batteries are not charged by newer laptops. I've this after changing laptop MB in Dell 3567 (I5 7th gen) to compatible mb from 3576 (i7 8th gen). And original battery stop charging until i by new one original with fresh manufacture date - there was changed battery data exchange protocol. Battery model still the same.
So, i will not surprise if the same situation can be with modern laptops and charger data exchange protocol by OneWire.

Laptops are good enough for power efficient home media server or even NAS, but need provide space clearance for ventilation anyway.
I want use my old one Dell 3585 case with MB from 5593 (i5 10th gen), it has 2 SATA ports useful for RAID-0/1 and one m2 pci-e port 2280.
Battery is good option to have UPS :)
